Src/AutoFixture.NUnit3.UnitTest/AutoDataAttributeTest.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Linq;
+using System.Runtime.InteropServices;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using NUnit.Framework.Internal;
+
+namespace Ploeh.AutoFixture.NUnit3.UnitTest
+{
+    [TestFixture]
+    public class AutoDataAttributeTest
+    {
+        [Test]
+        public void ExtendsAttribute()
+        {
+            Assert.That(new AutoDataAttribute(), Is.InstanceOf<Attribute>());
+        }
+
+        [Test]
+        public void ImplementsITestBuilder()
+        {
+            var autoDataFixture = new AutoDataAttribute();
+            Assert.That(autoDataFixture, Is.InstanceOf<ITestBuilder>());
+        }
+
+        [Test]
+        public void BuildFromYieldsRunnableTestMethod()
+        {
+            // Arrange
+            var autoDataAttribute = new AutoDataAttribute();
+            var fixtureType = this.GetType();
+
+            var methodWrapper = new MethodWrapper(fixtureType, fixtureType.GetMethod("DummyTestMethod"));
+            var testSuite = new TestSuite(fixtureType);
+
+            // Act
+            var testMethod = autoDataAttribute.BuildFrom(methodWrapper, testSuite).First();
+
+            // Assert
+            Assert.That(testMethod.RunState == RunState.Runnable);
+        }
+
+        /// <summary>
+        /// This is used in BuildFromYieldsParameterValues for building a unit test method
+        /// </summary>
+        public void DummyTestMethod(int anyInt, double anyDouble)
+        {
+        }
+
+        [Test]
+        public void CanBeExtendedToTakeAnIFixture()
+        {
+            var stub = new AutoDataAttributeStub(new DummyFixture());
+
+            Assert.That(stub, Is.AssignableTo<AutoDataAttribute>());
+        }
+
+        [Test]
+        public void IfCreateParametersThrowsExceptionThenReturnsNotRunnableTestMethodWithExceptionInfoAsSkipReason()
+        {
+            // Arrange
+            // DummyFixture is set up to throw DummyException when invoked by AutoDataAttribute
+            var autoDataAttributeStub = new AutoDataAttributeStub(new DummyFixture());
```

Comment:
The way `DummyFixture` is used here, the name is misleading. According to the terminology of [xUnit Test Patterns](http://bit.ly/xunitpatterns) (which is what we use in the AutoFixture code base), a _dummy_ is "a placeholder object that is passed to the SUT as an argument [...] but is never actually used."

The `DummyFixture` here, on the contrary, is used to provoke an exception; a _stub_ would be a more appropriate designation. I'd prefer calling it `ThrowingStubFixture` or something like that, in order to signify that it's going to throw an exception when used.